Sprint 4

LYNXhfdd opened the pull request.

LYNXhfdd committed:
Merge pull request #1 from LYNXhfdd/testBranch

изменён вывод в консоль в файле AppDelegate

LYNXhfdd committed:
Создан макет приложения в мэйн сториборде
Добавлена логика приложения
Добавлен уведомление о верном/неверном ответе
Добавлен алерт с результатами прохождения квиза и предложением сыграть ещё раз

LYNXhfdd committed:
Исправлен код вызова функции show во viewDidLoad

helecta commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,12 +1,184 @@
+    
+        
+        let currentQuestion = questions[currentQuestionIndex]
+        show(quiz: QuizStepViewModel(image: UIImage(named: currentQuestion.image) ?? UIImage(), question: currentQuestion.text, questionNumber: "\(currentQuestionIndex + 1)/\(questions.count)"))
```
у тебя есть метод, который преобразует QuizQuestion в QuizStepViewModel. Давай тут использовать его, иначе идет дублирование кода

helecta commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,12 +1,184 @@
+                self.show(quiz: viewModel)
+            }
+        // добавляем в алерт кнопки
+            alert.addAction(action)
```
Не забывай следить за отступами, общепринятые отступы помогают читать код быстрее. В общем случае поможет выделить весь файл cmd+A и зажать control+i.

helecta commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,12 +1,184 @@
+    }
+    
+    // для состояния "Вопрос задан"
+    private struct QuizStepViewModel {
```
Здорово, что добавляешь комментарии. Если добавлять через /// , а не // , то тогда можно будет быстро прочитать документацию, если зажать кнопку option и навести на класс/структуру/метод/свойство и т.д.

helecta commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,12 +1,184 @@
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+    }
+    
+    @IBAction private func noButtonClicked(_ sender: UIButton) {
```
Стоит организовывать код следующим образом:

- Публичные переменные
- IBOutlet
- Приватные переменные
- Публичные методы(сначала `init`ы, `override`ы, и потом остальные)
- Приватные методы
- @IBAction

Подробнее почитать можно тут: [Организация кода](https://github.com/RedMadRobot/RMR-swift-style-guide#code-organization)

LYNXhfdd committed:
Шрифты приведены в соответствие с макетом. Использованы цвета из макета. После выбора ответа через секунду автоматически появляется следующий вопрос и рамка ответа исчезает.
Теперь кнопки блокируются после выбора ответа во избежание случайных нажатий. Цвет фона изменён. Изменён цвет статус бара в приложении. Удалены лишние папки.

LYNXhfdd committed:
исправлен вызов функции show во viewDidLoad во избежание дублирования кода. Поправлена организация кода
